### PR TITLE
Support filament v3

### PR DESCRIPTION
**Pull request by dododedodonl**

### CODE REVIEW

**Comment by bramr94** [comment]
Known issues/todos:

- Error messages are not shown on the form after redirecting the user. https://github.com/DutchCodingCompany/filament-socialite/pull/46/files#diff-d5a8f85ccb80a2bcf426a5701223f57c80adf8e4a40b3837bd5d398ec4466711R59
- Readme needs to be updated.

**Comment by ChristophWolffNabu** [comment]
Any change this will be release soon?

**Comment by wannesmatthys** [comment]
Issues:
* When enabling Filament it's multi tenancy system (https://filamentphp.com/docs/3.x/panels/tenancy), there is no extra route parameter given with the current tenant, which results in an error after registering/logging in.
```Missing required parameter for [Route: filament.admin.pages.dashboard] [URI: {tenant}] [Missing parameter: tenant].```

* When using another panel ID than 'admin', it still uses admin as route, which results in this error:
```Route [filament.admin.pages.dashboard] not defined.```

**Comment by dododedodonl** [comment]
@wannesmatthys could you test the new version of this branch? The issues you mentioned should be resolved as far as we can test.

File: composer.json [hunk 1]
```diff
@@ -13,29 +13,35 @@
...
+            "role": "Developer"
         }
     ],
     "require": {
-        "php": "^8.0",
-        "filament/filament": "^2.0",
-        "illuminate/contracts": "^8.0|^9.0|^10.0",
+        "php": "^8.1",
+        "filament/filament": "^3.0",
+        "illuminate/contracts": "^9.0|^10.0",
```
Comment (dododedodonl) [inline composer.json:31]: Drop laravel 9 support

File: composer.json [hunk 2]
```diff
@@ -13,29 +13,35 @@
...
+            "email": "b.raaijmakers@hotmail.com",
+            "role": "Developer"
         }
     ],
     "require": {
-        "php": "^8.0",
-        "filament/filament": "^2.0",
-        "illuminate/contracts": "^8.0|^9.0|^10.0",
+        "php": "^8.1",
+        "filament/filament": "^3.0",
```
Comment (dododedodonl) [inline composer.json:30]: Update to `^3.0.9`

File: phpstan-baseline.neon [hunk 3]
```diff
@@ -0,0 +1,11 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Class App\\\\Models\\\\User not found\\.$#"
+			count: 1
+			path: config/filament-socialite.php
```
Comment (dododedodonl) [inline phpstan-baseline.neon:6]: This should be obsolete

File: .github/workflows/phpstan.yml [hunk 4]
```diff
@@ -1,30 +1,31 @@
...
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: true
       matrix:
-        os: [ubuntu-latest, windows-latest]
-        php: [8.1, 8.0]
-        laravel: [9.*]
+        os: [ubuntu-latest]
+        php: [8.2, 8.1]
+        laravel: [10.*, 9.*]
```
Comment (dododedodonl) [inline .github/workflows/phpstan.yml:17]: Drop laravel 9

**Pull request merged**